app/views/NotificationPreferencesView/index.tsx

```diff
@@ -1,282 +1,234 @@
-import React from 'react';
-import { StyleSheet, Switch, Text } from 'react-native';
-import { RouteProp } from '@react-navigation/core';
-import { StackNavigationProp } from '@react-navigation/stack';
-import { Observable, Subscription } from 'rxjs';
-import { connect } from 'react-redux';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/core';
+import React, { useEffect, useState } from 'react';
+import { Switch, Text } from 'react-native';
 
-import database from '../../lib/database';
-import { SWITCH_TRACK_COLOR, themes } from '../../lib/constants';
-import StatusBar from '../../containers/StatusBar';
+import { TActionSheetOptionsItem, useActionSheet } from '../../containers/ActionSheet';
+import { CustomIcon } from '../../containers/CustomIcon';
 import * as List from '../../containers/List';
-import I18n from '../../i18n';
-import { TSupportedThemes, withTheme } from '../../theme';
-import protectedFunction from '../../lib/methods/helpers/protectedFunction';
 import SafeAreaView from '../../containers/SafeAreaView';
+import StatusBar from '../../containers/StatusBar';
+import { IRoomNotifications, TRoomNotificationsModel } from '../../definitions';
+import I18n from '../../i18n';
+import { SWITCH_TRACK_COLOR } from '../../lib/constants';
+import { useAppSelector } from '../../lib/hooks';
+import { showErrorAlertWithEMessage } from '../../lib/methods/helpers';
+import { compareServerVersion } from '../../lib/methods/helpers/compareServerVersion';
 import log, { events, logEvent } from '../../lib/methods/helpers/log';
+import { Services } from '../../lib/services';
+import { ChatsStackParamList } from '../../stacks/types';
+import { useTheme } from '../../theme';
 import sharedStyles from '../Styles';
 import { IOptionsField, OPTIONS } from './options';
-import { ChatsStackParamList } from '../../stacks/types';
-import { IApplicationState, IRoomNotifications, TRoomNotificationsModel } from '../../definitions';
-import { Services } from '../../lib/services';
-import { compareServerVersion } from '../../lib/methods/helpers/compareServerVersion';
-
-const styles = StyleSheet.create({
-	pickerText: {
-		...sharedStyles.textRegular,
-		fontSize: 16
-	}
-});
 
-interface INotificationPreferencesViewProps {
-	navigation: StackNavigationProp<ChatsStackParamList, 'NotificationPrefView'>;
-	route: RouteProp<ChatsStackParamList, 'NotificationPrefView'>;
-	theme: TSupportedThemes;
-	serverVersion: string | null;
-}
-
-interface INotificationPreferencesViewState {
+interface IBaseParams {
+	preference: string;
```

Comment:
I think we could actually type this.
This preference is an abstraction of both `OPTIONS` or `IRoomNotifications`.

For the first one, you could
- use `keyof typeof OPTIONS`
- remove [this nonsense](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/60f781d20e24ca92f9cd2c5dd9ef41de16fd0767/app/views/NotificationPreferencesView/options.ts#L7)
- remove the assertion `as IOptionsField` 
- Everything `RenderListPicker` is going to be inferred

Apply something like that to `IRoomNotifications` and we're good.